portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,21 +26,33 @@
 import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
+  // comments data structure
+  ArrayList<String> comments = new ArrayList();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> list = new ArrayList<>();
-    list.add("first message");
-    list.add("second message");
-    list.add("third message");
-
     Gson gson = new Gson();
-    String json = gson.toJson(list);
-
+    String json = gson.toJson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("text_input");
+    String name = request.getParameter("name_input");
+    if (comment == null || name == null) {
+      response.sendRedirect("/index.html");  
```

Comment:
Nit: An interesting stretch goal could be to provide some feedback to the user (whether it failed or not).